pandas/tests/copy_view/test_setitem.py

```diff
@@ -58,18 +58,18 @@ def test_set_column_with_index(using_copy_on_write):
     # the index data is copied
     assert not np.shares_memory(get_array(df, "c"), idx.values)
 
-    # and thus modifying the index does not modify the DataFrame
-    idx.values[0] = 0
-    tm.assert_series_equal(df["c"], Series([1, 2, 3], name="c"))
-
     idx = RangeIndex(1, 4)
     arr = idx.values
 
     df["d"] = idx
 
     assert not np.shares_memory(get_array(df, "d"), arr)
-    arr[0] = 0
-    tm.assert_series_equal(df["d"], Series([1, 2, 3], name="d"))
+    if using_copy_on_write:
+        with pytest.raises(ValueError, match="assignment"):
+            arr[0] = 0
+    else:
+        arr[0] = 0
+        tm.assert_series_equal(df["d"], Series([1, 2, 3], name="d"))
```

Comment:
```suggestion
```

(similarly as the case above that you updated, can just remove this, since we already test with `shares_memory`)